packages/api/amplify_api/lib/src/util.dart

```diff
@@ -0,0 +1,29 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:meta/meta.dart';
+
+/// Sets the 'Content-Type' of headers to match the [HttpPayload] body.
+@internal
+Map<String, String>? addContentTypeToHeaders(
+    Map<String, String>? headers, HttpPayload? body) {
+  var modifiedHeaders = headers != null ? Map.of(headers) : null;
```

Comment:
I see why this is needed, but a comment and/or test to state as much would be helpful for others, since I don't think it's obvious why we can't modify the headers directly

Reply:
added comment and test